File: AMS.API/DTO/Assessor/SearchAssessorRequestParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace AMS.API.DTO.Assessor
{
    public class SearchAssessorRequestParams : AssessorRequestParams
    {
        public string Name { get; set; }

        public char? Gender { get; set; }

        public string EmailId { get; set; }

        public string MobileNo { get; set; }

        public string WhatsAppNo { get; set; }

        public int StateId { get; set; }

        public int CityId { get; set; }

        #region Additional Advance Params

        public int SkillCouncilTypeId { get; set; }

        public int SkillCouncilId { get; set; }

        public int JobRoleId { get; set; }

        public int IdProofTypeId { get; set; }

        public string IdProofValue { get; set; }

        public string AccountNumber { get; set; }

        #endregion
    }
}

File: AMS.Web/src/app/models/assessor-demographic/search-assessor-request-params.ts
export class SearchAssessorRequestParams 
{
    //primary details
    AssessorId: number = null;
    IRIS_Id: string = null;
    SSC_Id: string = null;
    NSDC_Id: string = null;

    //other primary search params
    Name?: string = null;
    Gender?: string = null;
    EmailId?: string = null;
    MobileNo?: string = null;
    WhatsAppNo?: string = null;
    StateId?: number = null;
    CityId?: number = null;

    //#region all other Advance search params
    SkillCouncilTypeId: number = null;
    SkillCouncilId: number = null;
    JobRoleId: number = null;
    IdProofTypeId: number = null;
    IdProofValue?: string = null;
    AccountNumber?: string = null;
    //#endregion 

    IsActive: boolean = true;
}
